src/test/java/io/pinecone/PineconeIndexOperationsTest.java

```diff
@@ -0,0 +1,212 @@
+package io.pinecone;
+
+import com.google.gson.Gson;
+import io.pinecone.clients.Pinecone;
+import io.pinecone.exceptions.PineconeValidationException;
+import okhttp3.*;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.openapitools.client.model.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.times;
+
+public class PineconeIndexOperationsTest {
+    private static final Gson gson = new Gson();
+
+    @Test
+    public void testDeleteIndex() throws IOException {
+        Call mockCall = mock(Call.class);
+        when(mockCall.execute()).thenReturn(new Response.Builder()
+                .request(new Request.Builder().url("http://localhost").build())
+                .protocol(Protocol.HTTP_1_1)
+                .code(200)
+                .message("OK")
+                .body(ResponseBody.create("Response body", MediaType.parse("text/plain")))
+                .build());
+        OkHttpClient mockClient = mock(OkHttpClient.class);
+        when(mockClient.newCall(any(Request.class))).thenReturn(mockCall);
+
+        Pinecone client = new Pinecone.Client("testAPiKey").withOkHttpClient(mockClient).build();
+        client.deleteIndex("testIndex");
+
+        verify(mockClient, times(1)).newCall(any(Request.class));
+        verify(mockCall, times(1)).execute();
+    }
```

Comment:
If I'm reading the validation correctly here, it's just checking that *some/any* request was called. Is there a way we can check whether the delete index request was made?

Reply:
I think at the time of coding this, my logic was there should exactly be one request made and since its client.deleteIndex, we assumed its delete index. But that's a good point.

Reply:
I think we can separate the potential test changes in a follow-up PR

Reply:
Made some updates to this to validate a bit more granularly. We're now using `ArgumentCaptor` to capture information about the request and validate the HTTP method and URL. There's probably more we can do but this is at least specific about what it's validating for.